### DISCUSSION TITLE
Table column does not get detected

### DISCUSSION THREAD

**Discussion by malek-ewa** [jsvine/pdfplumber#846] [Ask for help with specific PDFs]
Hey there :)

I've been trying to use the "extract_table" method to extract a table from a PDF file. For some reason, this works on the first page of the file, but on the rest of the pages, the last column of the table is not getting extracted at all.

This is how the first page of my PDF looks like: [click](https://www.evernote.com/shard/s631/sh/b82ca91e-2f7b-41c7-bc5c-1c6bebbb7162/PPLHg2uVEkgq998j2ZTv8G7A7u8_sKzvBw4YKSH672ra1ZE6_l4D7Kjm4A)
And this is how the second page looks like: [click](https://www.evernote.com/shard/s631/sh/858d7816-f334-4cc5-bc72-9df17dd828f2/PQP0FUcTdppYkoxQGzEimKq5EmfhHi7Ft-MogsoHNgptWYucfYLtf63evQ)

I am running the below code on both pages:

```
with pdfplumber.open(filepath) as pdf:
    page = pdf.pages[0] # or 1 if it's the second page
    table = pd.DataFrame(
        page.extract_table(table_settings={
            "keep_blank_chars" : True,
            "horizontal_strategy": "text",
            "snap_y_tolerance": 4
        })
    )
print(table.head(10))
```

For the first page, I'm getting this: [screenshot here](https://www.evernote.com/shard/s631/sh/c5560fd0-9ebc-4e20-9796-563dc04298a9/bdncWAGu58LR_zXDB6ZWQ54HouqsMR_mWZhDNCT4BASUFWQMqhkIeoOYkA)
But for the second page, I'm getting this: [screenshot here](https://www.evernote.com/shard/s631/sh/36f33f63-38f1-467d-8388-155abfe36d10/on0QxJJKAwJh5LBXlvnFvDhQJONx6FoUG_eTED8oibmzhtl8ZrTtnJMliw)
→ You can see in the 2nd screenshot that the last column is missing.

I tried adding `page = page.crop((0, 0, page.width, page.height))` after `page = pdf.pages[0]` (wa thinking this might force the code to look at the whole PDF page, and thus see the last column), but that did not help much

I also tried different sets of parameters, one that delivered an interesting output is:
```
with pdfplumber.open(filepath) as pdf:
    page = pdf.pages[1]
    table = pd.DataFrame(page.extract_table(table_settings={
        "keep_blank_chars" : True,
        "horizontal_strategy": "text",
        "snap_y_tolerance": 4,
        "vertical_strategy": "text"
    }))
```
With these settings, the last column is captured, but the first 2 columns get a bit messed up ([see screenshot of the output here](https://www.evernote.com/shard/s631/sh/3976c734-999a-45cd-8e86-8f0bdd341990/075pY8causde_Ondm-zsHOZj5z0SsbLWyf0__Cake07rpY99sYtsNdCp-Q))

Could someone please advise on why I'm having this issue and how I could get it resolved ?

**PS:** Since the files contain sensitive information, I preferred sharing screenshots only to start with, in case this is a known issue and that you already have some suggestions on what I could try (even without having the real PDF files).

Thanks in advance.

**Comment by jsvine** [comment]
Hi @malek-ewa, without access to the PDF, it's hard to know what's happening for sure. A couple of suggestions based on the screenshots, however:

- Try using `pdfplumber`'s visual debugging tools (described in the README), and in particular `PageImage.debug_tablefinder(settings)`. That'll give you a better sense of what `pdfplumber` is doing behind the scenes.
- Try setting `intersection_x_tolerance` to a relatively high value, such as 10 or 20, which can be helpful when using `"horizontal_strategy": "text"`.

**Reply by malek-ewa** [reply]
Hi @jsvine . Thank you very much for the prompt response.
Setting `intersection_x_tolerance` to 20 solved my issue actually :D
Thank you so much for your support
